### DISCUSSION TITLE
An attempt at comparing io_uring based eventloop to epoll based one rigorously.

### DISCUSSION THREAD

**Discussion by kyoung-seo** [axboe/liburing#1299] [General]
I have been working on characterizing performance of io_uring by porting event loop of memcached and varnish to utilize io_uring.

I am sharing some results here [0].

I am curious if numbers I observed is inline with other people's experience and intuition with io_uring.

[0] https://git.uwaterloo.ca/lseo/io_uring-experiments

**Comment by cmazakas** [comment]
I've observed somewhat similar numbers. I have a benchmark comparing me to Asio and I was only 18% faster which isn't _too_ far out of line with your results. Once I added timeout algorithms, however, I was dramatically faster.

**Reply by kyoung-seo** [reply]
Hello Christian,

Thanks for your input. By timeout algorithm, did you mean you are dynamically adjusting batch size/timeout when reaping CQEs? Could you please describe what you did? I am very interested in how people are adjusting batch parameters.

**Comment by isilence** [comment]
> I have been working on characterizing performance of io_uring by porting event loop of memcached and varnish to utilize io_uring.
> 
> I am sharing some results here [0].
> 
> I am curious if numbers I observed is inline with other people's experience and intuition with io_uring.
> 
> [0] https://git.uwaterloo.ca/lseo/io_uring-experiments

Thanks for sharing, it looks like a solid comparison from a quick look. Pokes interest to stare at numbers for a bit longer and probably check if the test is doing everything optimally, but I'd need to find time for it.

**Reply by kyoung-seo** [reply]
Hello Pavel,

Thank you for taking a look at it.

I followed most of the advice in here [0] but I might have made a suboptimal decisions in some places and there are some parameters I am still not sure how to fine tune like batch size or buffer size.

I would appreciate any kind of feedback when you get a chance to look at it :)

[0] https://github.com/axboe/liburing/wiki/io_uring-and-networking-in-2023
